internal/analyzers/prometheus.go

```diff
@@ -0,0 +1,271 @@
+        fmt.Printf("Prometheus matches all namespaces\n")
+    }
+
+	if err := checkPrometheusServiceSelectorsStatus(ctx, clientSets, prometheus.Spec.ServiceMonitorSelector, "ServiceMonitor", namespace); err != nil {
+		return fmt.Errorf("ServiceMonitor is not properly defined: %s", err)
+	}
+
+	if err := checkPrometheusServiceSelectorsStatus(ctx, clientSets, prometheus.Spec.PodMonitorSelector, "PodMonitor", namespace); err != nil {
+		return fmt.Errorf("PodMonitor is not properly defined: %s", err)
+	}
+
+	if err := checkPrometheusServiceSelectorsStatus(ctx, clientSets, prometheus.Spec.ProbeSelector, "Probe", namespace); err != nil {
+		return fmt.Errorf("Probe is not properly defined: %s", err)
+	}
+
+	if err := checkPrometheusServiceSelectorsStatus(ctx, clientSets, prometheus.Spec.ScrapeConfigSelector, "ScrapeConfig", namespace); err != nil {
+		return fmt.Errorf("ScrapeConfig is not properly defined: %s", err)
+	}
+
+	slog.Info("Prometheus is compliant, no issues found", "name", name, "namespace", namespace)
+	return nil
+}
+
+func checkClusterRoleRules(crb v1.ClusterRoleBinding, cr *v1.ClusterRole) error {
+	var errs []string
+	verbsToCheck := []string{"get", "list", "watch"}
+	missingVerbs := []string{}
+
+	for _, rule := range cr.Rules {
+		for _, resource := range rule.Resources {
+			found := false
+			if resource == "configmaps" {
+				for _, verb := range rule.Verbs {
+					if verb == "get" {
+						found = true
+						break
+					}
+				}
+				if !found {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'configmaps' with 'get' in its verbs", crb.RoleRef.Name))
+				}
+				continue
+			}
+			for range rule.APIGroups {
+				for _, requiredVerb := range verbsToCheck {
+					found := false
+					for _, verb := range rule.Verbs {
+						if verb == requiredVerb {
+							found = true
+							break
+						}
+					}
+					if !found {
+						missingVerbs = append(missingVerbs, requiredVerb)
+					}
+				}
+				if len(missingVerbs) > 0 {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s is missing necessary verbs for APIGroups: %v", crb.RoleRef.Name, missingVerbs))
+				}
+			}
+		}
+		for _, nonResource := range rule.NonResourceURLs {
+			if nonResource == "/metrics" {
+				hasGet := false
+				for _, verb := range rule.Verbs {
+					if verb == "get" {
+						hasGet = true
+						break
+					}
+				}
+				if !hasGet {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'get' verb for NonResourceURL '/metrics'", crb.RoleRef.Name))
+				}
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("multiple errors found:\n%s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func checkPrometheusNamespaceSelectorsStatus(ctx context.Context, clientSets *k8sutil.ClientSets, labelSelector *metav1.LabelSelector) (bool, error) {
+    if labelSelector == nil {
+        return true, nil
+    }
+
+    if len(labelSelector.MatchLabels) == 0 && len(labelSelector.MatchExpressions) == 0 {
```

Comment:
I think is this condition is true we don't thing we need to check if we can list namespaces, by default kubernetes will have namespaces like default and kube-system